### PR TITLE
ZEN-4357 Evaluate Guava dependencies, consider removing where practical

### PR DESCRIPTION
**Pull request by tfesenko**
Guava is a recurring problem with Eclipse. We should consider removing our use of Guava, instead using FP and other features of Java 8.

I removed usages of Guava in KZOE, but I still need to keep a dependency on Guava in our manifest declaration because [JSON Schema Validator uses Guava](https://github.com/java-json-tools/json-schema-core/blob/master/src/main/java/com/github/fge/jsonschema/core/load/configuration/URIDownloadersRegistry.java)

### CODE REVIEW

**Review by andylowry — APPROVED** [review]
Whew! Looks good.

File: com.reprezen.swagedit.core/src/com/reprezen/swagedit/core/assist/JsonContentAssistProcessor.java [hunk 1]
```diff
@@ -196,7 +195,7 @@ protected void clearStatus() {
             int offset) {
         final List<ICompletionProposal> result = new ArrayList<>();
 
-        prefix = Strings.emptyToNull(prefix);
+        prefix = "".equals(prefix) ? null : prefix;
```
Comment (ghillairet) [inline com.reprezen.swagedit.core/src/com/reprezen/swagedit/core/assist/JsonContentAssistProcessor.java]: We should be using `StringUtils.emptyToNull` here.

**Review by ghillairet — APPROVED** [review]
@tfesenko Looks good. I made 2 commits to solve an issue with Arrays.asList(). This method creates an immutable list, and we were using it to create lists that are then updated. 

It also removes guava from tests MANIFEST and target platform as it should not be needed as Guava comes with Xtend.

File: com.reprezen.swagedit.core/src/com/reprezen/swagedit/core/assist/JsonContentAssistProcessor.java (hunk 1 above)
Comment (ghillairet) [reply com.reprezen.swagedit.core/src/com/reprezen/swagedit/core/assist/JsonContentAssistProcessor.java]: @tfesenko I did the change.

**Comment by ghillairet** [comment]
@tfesenko Please review my 2 commits and merge if you think it's ok.

File: com.reprezen.swagedit.core/src/com/reprezen/swagedit/core/editor/outline/OutlineContentProvider.java [hunk 2]
```diff
@@ -48,7 +48,7 @@ public void inputChanged(Viewer viewer, Object oldInput, Object newInput) {
             return null;
         }
 
-        List<Object> roots = Arrays.asList();
+        List<Object> roots = new ArrayList<>();
```
Comment (tfesenko) [inline com.reprezen.swagedit.core/src/com/reprezen/swagedit/core/editor/outline/OutlineContentProvider.java]: Thanks!

**Pull request merged**